framework/src/actions/CreateExecutionerAction.C

```diff
@@ -110,6 +110,7 @@ CreateExecutionerAction::act()
   Moose::setup_perf_log.pop("Create Executioner","Setup");
 
   _awh.executioner() = executioner;
```

Comment:
Actually `Action` constructs a reference of the shared pointer from AWH. If I get rid of AWH accessor, I will need to change MooseApp interface, which will result into a painful set of changes, possibly patching applications. So do we really want to remove the executioner accesso in AWH?